src/dataflow/src/render/upsert.rs

```diff
@@ -462,7 +486,9 @@ where
                                                 res.map(|v| {
                                                     rehydrate(
                                                         &upsert_envelope.key_indices,
-                                                        &decoded_key,
+                                                        // The value is never Ok
+                                                        // unless the key is also
+                                                        &decoded_key.as_ref().unwrap(),
```

Comment:
I think the `&` is not needed anymore because `as_ref()` already gets us a reference. Same below.

Reply:
good point!